cheru94/ranty-ebased-training
# feat(integration): added CRUD client endpoint, operation endpoint & migrated lambdas

## What does this PR do?

To begin this PR includes several features, which the main of them are detailed below:

- **CRUD** for **Client** domain. Including ( _Create [POST], Delete [PUT], Get All  [GET],  Get by ID  [GET], Update_ [PUT] )
- **[POST]** endpoint for **Operation** domain which validates the **existence** & the **status** of the **Client**
- Infrastructure as code with propper scaffolding

## Notes

To test purpose, my suggestion is to consume the API Gateway Endpoints with Postman, and use the Serverless CLI to watch the logs



## How To Test

1. Checkout this branch
2.  Deploy the infrastructure with `npm run deployIntegration`
3. Open Postman and check the response

## Screenshots

### Create Client

#### 200 OK

![image](https://user-images.githubusercontent.com/8229412/124528226-0f0e3680-ddde-11eb-83a0-a6c3cb71bf66.png)

#### 400 BAD REQUEST
![image](https://user-images.githubusercontent.com/8229412/124528181-ec7c1d80-dddd-11eb-88ca-c41136d7c870.png)

### Get All Clients
![image](https://user-images.githubusercontent.com/8229412/124528363-62808480-ddde-11eb-8f08-015d567ccf56.png)


### Get By Id Client

![image](https://user-images.githubusercontent.com/8229412/124528381-70cea080-ddde-11eb-834a-be7756d99494.png)


### Delete Client

![image](https://user-images.githubusercontent.com/8229412/124528400-77f5ae80-ddde-11eb-822a-a426f5d8b1a4.png)

### Update Client
![image](https://user-images.githubusercontent.com/8229412/124528427-82b04380-ddde-11eb-9e57-734218842d06.png)


### Operation

#### 200 OK
![image](https://user-images.githubusercontent.com/8229412/124528456-92c82300-ddde-11eb-8f0d-50ab9291d4a4.png)

#### 400 BAD REQUEST
![image](https://user-images.githubusercontent.com/8229412/124528528-c2772b00-ddde-11eb-95b8-0b05d48b81de.png)
